src/main/java/com/microsoft/sqlserver/jdbc/SQLServerDriver.java

```diff
@@ -903,11 +903,30 @@ static String getPropertyOnlyName(String name, Logger logger) {
         return null;
     }
 
+    private final static String[] systemPropertiesToLog = new String[] { "java.specification.vendor",
+            "java.specification.version", "java.class.path", "java.class.version", "java.runtime.name",
+            "java.runtime.version", "java.vendor", "java.version", "java.vm.name", "java.vm.vendor", "java.vm.version",
+            "java.vm.specification.vendor", "java.vm.specification.version", "os.name", "os.version", "os.arch" };
+
     @Override
     public java.sql.Connection connect(String Url, Properties suppliedProperties) throws SQLServerException {
         loggerExternal.entering(getClassNameLogging(), "connect", "Arguments not traced.");
         SQLServerConnection result = null;
 
+        if (loggerExternal.isLoggable(Level.FINE)) {
+            loggerExternal.log(Level.FINE, "Microsoft JDBC Driver " + SQLJdbcVersion.major + "." + SQLJdbcVersion.minor +
+                    "." + SQLJdbcVersion.patch + "." + SQLJdbcVersion.build + SQLJdbcVersion.releaseExt + " for SQL Server");
+            if (loggerExternal.isLoggable(Level.FINER)) {
+                for (String propertyKeyName : systemPropertiesToLog) {
+                    String propertyValue = System.getProperty(propertyKeyName);
+                    if (propertyValue != null && !propertyValue.isEmpty()) {
```

Comment:
```suggestion
                    if (null != propertyValue && !propertyValue.isEmpty()) {
```

Reply:
That style honestly drives me nuts. It just reads unnaturally to me. I've read the arguments for both styles, and there is no real advantage to one over the other in Java, that I'm aware of.

Our [Coding Guidelines](https://github.com/microsoft/mssql-jdbc/blob/dev/Coding_Guidelines.md) don't mention it, either. Is this an unspoken style the team follows?